build.sbt

```diff
@@ -459,6 +460,25 @@ lazy val docs = project // new documentation project
     )
   )
   .settings(fatalWarningsSettings: _*)
+  .settings(
+    firtoolVersionsTableTask / fileInputs ++= {
+      val rootGlob = (root / baseDirectory).value.toGlob
+      Seq(rootGlob / "build.sbt", rootGlob / "project" / "*.sbt", rootGlob / "project" / "*.scala")
+    },
+    firtoolVersionsTableTask := {
+      val logger = streams.value.log
+      val file = (Compile / sourceManaged).value / "FirtoolVersionsTable.scala"
+      // Only write the file if an input has changed
+      if (!file.exists || firtoolVersionsTableTask.inputFileChanges.hasChanges) {
+        // Escaping newlines makes it easier to generate the file
+        val table = FirtoolVersionsTable.generateTable.replaceAll("\n", "\\\\n")
+        logger.info(s"Writing $file...")
+        IO.write(file, s"""object FirtoolVersionsTable { def table = "$table" }""")
+      }
+      Seq(file)
+    },
+    Compile / sourceGenerators += firtoolVersionsTableTask.taskValue
+  )
```

Comment:
This also illustrates how to generate files in SBT and have it properly track dependencies, was a bit frustrating to figure out but not bad once you have it working.